rtl/eclrtl/rtlfield.cpp

```diff
@@ -514,100 +514,84 @@ bool RtlIntTypeInfo::canExtend(char &fillChar) const
+size32_t RtlBlobTypeInfo::process(const byte * self, const byte * selfrow, const RtlFieldInfo * field, IFieldProcessor & target) const
 {
     assertex(callback);
-    if (isUnsigned())
-        target.processUInt(callback->getFilePosition(nullptr), field);
-    else
-        target.processInt(callback->getFilePosition(nullptr), field);
-    return length;
+    throwUnexpected();
 }
 
-size32_t RtlFileposTypeInfo::toXML(const byte * self, const byte * selfrow, const RtlFieldInfo * field, IXmlWriter & target) const
+size32_t RtlBlobTypeInfo::toXML(const byte * self, const byte * selfrow, const RtlFieldInfo * field, IXmlWriter & target) const
 {
     assertex(callback);
-    if (isUnsigned())
-        target.outputUInt(callback->getFilePosition(nullptr), length, queryScalarXPath(field));
-    else
-        target.outputInt(callback->getFilePosition(nullptr), length, queryScalarXPath(field));
-    return length;
+    throwUnexpected();
 }
 
-void RtlFileposTypeInfo::getString(size32_t & resultLen, char * & result, const void * ptr) const
+void RtlBlobTypeInfo::getString(size32_t & resultLen, char * & result, const void * ptr) const
 {
     assertex(callback);
-    if (isUnsigned())
-        rtlUInt8ToStrX(resultLen, result, callback->getFilePosition(nullptr));
-    else
-        rtlInt8ToStrX(resultLen, result, callback->getFilePosition(nullptr));
+    throwUnexpected();
 }
 
-void RtlFileposTypeInfo::getUtf8(size32_t & resultLen, char * & result, const void * ptr) const
+void RtlBlobTypeInfo::getUtf8(size32_t & resultLen, char * & result, const void * ptr) const
 {
-    getString(resultLen, result, ptr);
+    throwUnexpected();
 }
 
-__int64 RtlFileposTypeInfo::getInt(const void * ptr) const
+__int64 RtlBlobTypeInfo::getInt(const void * ptr) const
 {
-    assertex(callback);
-    return (__int64) callback->getFilePosition(nullptr);
+    throwUnexpected();
 }
 
-double RtlFileposTypeInfo::getReal(const void * ptr) const
+double RtlBlobTypeInfo::getReal(const void * ptr) const
 {
-    assertex(callback);
-    if (isUnsigned())
-        return (double) (__uint64) callback->getFilePosition(nullptr);
-    else
-        return (double) (__int64) callback->getFilePosition(nullptr);
+    throwUnexpected();
 }
 
-bool RtlFileposTypeInfo::canTruncate() const
+bool RtlBlobTypeInfo::canTruncate() const
 {
     return false;
 }
 
-bool RtlFileposTypeInfo::canExtend(char &fillChar) const
+bool RtlBlobTypeInfo::canExtend(char &fillChar) const
 {
     return false;
 }
 
-void RtlFileposTypeInfo::setCallback(IThorIndexCallback *_callback)
+void RtlBlobTypeInfo::setCallback(IThorIndexCallback *_callback)
 {
     callback = _callback;
 }
 
-int RtlFileposTypeInfo::compare(const byte * left, const byte * right) const
+int RtlBlobTypeInfo::compare(const byte * left, const byte * right) const
 {
     throwUnexpected();  // Not needed and unimplementable
```

Comment:
Is needed, if postfiltering...

Reply:
Yes, they should all be UNIMPLEMENTED - I left it as work to be done in the future.

Reply:
Comment is wrong then...